src/program.hpp

```diff
@@ -418,23 +419,27 @@ struct cvk_program;
 
 class cvk_entry_point {
 public:
-    cvk_entry_point(VkDevice dev, cvk_program* program,
+    cvk_entry_point(cvk_device* dev, cvk_program* program,
                     const std::string& name);
 
     ~cvk_entry_point() {
         for (auto pipeline : m_pipelines) {
             cvk_info("destroying pipeline %p for kernel %s", pipeline.second,
                      m_name.c_str());
-            vkDestroyPipeline(m_device, pipeline.second, nullptr);
+            vkDestroyPipeline(m_device->vulkan_device(), pipeline.second,
```

Comment:
You could define a `vkdev` variable to shorthen those lines. It's a common idiom throughout the code base.